README.md

```diff
@@ -1,12 +1,225 @@
 # application-aware-quota
 Empower users to customize and manage resource quotas per component in Kubernetes, utilizing the scheduling gates feature.
 
+## Motivation
+In a multi-tenant cluster where multiple users or teams share a Kubernetes cluster, 
+there is a risk that one team might consume more than its fair share of resources. 
+Resource quotas are a tool for administrators to manage and mitigate this concern.
+
+Native ResourceQuotas in Kubernetes are designed to handle predefined pod resources such as
+`requests.cpu`, `limits.cpu`, `requests.memory`, `limits.memory`, and more. This architecture
+wasn't designed to be easily extendable.
+
+Native pod resources are often considered implementation details that can change across different development 
+stages or versions of Kubernetes Operators.
+
+Due to ResourceQuota being unextendable they are not suited for more complex scenarios, such as limiting 
+resources based on the broader application's context rather than solely on the pod's resource requirements. 
+With AAQ, it's possible to not take into account the pod's resource requirements into consideration at all, 
+but instead base it on other, possible multiple factors such as different pod(s) fields, CRD fields, and more.
+
+Another issue is that the native ResourceQuotas in Kubernetes are admission-based, which can
+cause the resource quota admission controller to become a bottleneck for the API Server during
+evaluation and ResourceQuota status update calls. This potential burden on the API server further
+exacerbates the challenge of extending the design, such as implementing Multi-Namespace Quota.
+
+Additionally, if a quota is set with `requests.cpu`, `limits.cpu`, `requests.memory`, or
+`limits.memory` boundaries in a namespace, then pods that don't set these resources are not
+allowed to be created. This essentially means that in such namespaces, best effort pods and some
+burstable pods are not allowed. This limitation in Kubernetes was a mistake that currently exists due
+to compatibility reasons as mentioned in the following
+[link](https://github.com/kubernetes/kubernetes/blob/c876b30c2b30c0355045d7548c22b6cd42ab58da/pkg/quota/v1/evaluator/core/pods.go#L89).
+
+## User Stories
+
+**User Story**: Managing VM Billing and Resource Quotas in a Multi-Tenant Cluster with KubeVirt
+
+As an administrator of a multi-tenant cluster with various operators installed 
+, including the [KubeVirt Operator](https://github.com/kubevirt/kubevirt?tab=readme-ov-file#introduction) 
```

Comment:
This looks out of place. Perhaps move the leading comma to the previous line?

Reply:
Done.